src/Caching/StackExchangeRedis/src/RedisOutputCacheStore.cs

```diff
@@ -0,0 +1,487 @@
+
+    // Never reconnect within 60 seconds of the last attempt to connect or reconnect.
+    private readonly TimeSpan ReconnectMinInterval = TimeSpan.FromSeconds(60);
+    // Only reconnect if errors have occurred for at least the last 30 seconds.
+    // This count resets if there are no errors for 30 seconds
+    private readonly TimeSpan ReconnectErrorThreshold = TimeSpan.FromSeconds(30);
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="RedisCache"/>.
+    /// </summary>
+    /// <param name="optionsAccessor">The configuration options.</param>
+    public RedisOutputCacheStore(IOptions<RedisCacheOptions> optionsAccessor) // TODO: OC-specific options?
+        : this(optionsAccessor, Logging.Abstractions.NullLoggerFactory.Instance.CreateLogger<RedisCache>())
+    {
+    }
+
+#if DEBUG
+    internal async ValueTask<string> GetConfigurationInfoAsync(CancellationToken cancellationToken = default)
+    {
+        await ConnectAsync(cancellationToken).ConfigureAwait(false);
+        return $"redis output-cache; MULTI/EXEC: {_useMultiExec}, v6.2+: {_use62Features}";
+    }
+#endif
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="RedisCache"/>.
+    /// </summary>
+    /// <param name="optionsAccessor">The configuration options.</param>
+    /// <param name="logger">The logger.</param>
+    internal RedisOutputCacheStore(IOptions<RedisCacheOptions> optionsAccessor, ILogger logger)
+    {
+        ArgumentNullThrowHelper.ThrowIfNull(optionsAccessor);
+        ArgumentNullThrowHelper.ThrowIfNull(logger);
+
+        _options = optionsAccessor.Value;
+        _logger = logger;
+
+        // This allows partitioning a single backend cache for use with multiple apps/services.
+
+        // SE.Redis allows efficient append of key-prefix scenarios, but we can help it
+        // avoid some work/allocations by forcing the key-prefix to be a byte[]; SE.Redis
+        // would do this itself anyway, using UTF8
+        _valueKeyPrefix = (RedisKey)Encoding.UTF8.GetBytes(_options.InstanceName + "__MSOCV_");
+        _tagKeyPrefix = (RedisKey)Encoding.UTF8.GetBytes(_options.InstanceName + "__MSOCT_");
+        _tagMasterKey = (RedisKey)Encoding.UTF8.GetBytes(_options.InstanceName + "__MSOCT");
+        _tagMasterKeyArray = new[] { _tagMasterKey };
+
+        _ = Task.Factory.StartNew(RunGarbageCollectionLoopAsync, default, TaskCreationOptions.LongRunning, TaskScheduler.Current);
+    }
+
+    private async Task RunGarbageCollectionLoopAsync()
+    {
+        try
+        {
+            while (!Volatile.Read(ref _disposed))
+            {
+                // approx every 5 minutes, with some randomization to prevent spikes of pile-on
+                var secondsWithJitter = 300 + Random.Shared.Next(-30, 30);
+                Debug.Assert(secondsWithJitter >= 270 && secondsWithJitter <= 330);
+                await Task.Delay(TimeSpan.FromSeconds(secondsWithJitter)).ConfigureAwait(false);
```

Comment:
Consider using `PeriodicTimer`, it would also let you stop the timer in dispose so we don't have a long running task in the background during shutdown.